projects/components/src/filter-bar/filter/parser/filter-parser.service.ts

```diff
@@ -24,7 +24,8 @@ export class FilterParserService {
   public constructor(private readonly filterBuilderService: FilterBuilderService) {}
 
   public parseUserFilterString(filterString: string, attribute: FilterAttribute): Filter | undefined {
-    const parsed = this.parseFilterString(filterString, attribute, USER_FILTER_OPERATORS, attribute.displayName);
+    const spacedOperators = USER_FILTER_OPERATORS.map(op => ` ${op} `);
```

Comment:
why is the space needed?

Reply:
Since we intend to split on the operator. In this case is "in", which is also often in values and keys.

The filter pills enforce a space around the operator so this will work for now, but I'm not a fan of relying on the component for validation. This feels too brittle. We really need a full on parser or there are going to be corner cases we miss.

Reply:
If only we had a parsing library.. :)

Reply:
I still need to research more, but from what I saw when I started investigating, we need a little more than parjs as it missed the same use cases my own string parsing did.